packages/automerge-repo/src/DocHandle.ts

```diff
@@ -255,157 +174,202 @@ export class DocHandle<T> //
+   * received the document for the first time, signal that our request has been completed.
+   */
+  #checkForChanges(before: T | undefined, after: T) {
+    const docChanged =
+      after && before && !headsAreSame(A.getHeads(after), A.getHeads(before))
+    if (docChanged) {
+      this.emit("heads-changed", { handle: this, doc: after })
+
+      const patches = A.diff(after, A.getHeads(before), A.getHeads(after))
+      if (patches.length > 0) {
+        this.emit("change", {
+          handle: this,
+          doc: after,
+          patches,
+          // TODO: pass along the source (load/change/network)
+          patchInfo: { before, after, source: "change" },
+        })
+      }
+
+      // If we didn't have the document yet, signal that we now do
+      if (!this.isReady()) this.#machine.send({ type: DOC_READY })
+    }
+    this.#prevDocState = after
+  }
+
   // PUBLIC
 
+  /** Our documentId in Automerge URL form.
+   */
+  get url(): AutomergeUrl {
+    return stringifyAutomergeUrl({ documentId: this.documentId })
+  }
+
   /**
-   * Checks if the document is ready for accessing or changes.
-   * Note that for documents already stored locally this occurs before synchronization
-   * with any peers. We do not currently have an equivalent `whenSynced()`.
+   * @returns true if the document is ready for accessing or changes.
+   *
+   * Note that for documents already stored locally this occurs before synchronization with any
+   * peers. We do not currently have an equivalent `whenSynced()`.
+   */
+  isReady = () => this.inState(["ready"])
+
+  /**
+   * @returns true if the document has been marked as deleted.
+   *
+   * Deleted documents are removed from local storage and the sync process. It's not currently
+   * possible at runtime to undelete a document.
+   */
+  isDeleted = () => this.inState(["deleted"])
+
+  /**
+   * @returns true if the document is currently unavailable.
+   *
+   * This will be the case if the document is not found in storage and no peers have shared it with us.
    */
-  isReady = () => this.inState([HandleState.READY])
+  isUnavailable = () => this.inState(["unavailable"])
+
   /**
-   * Checks if this document has been marked as deleted.
-   * Deleted documents are removed from local storage and the sync process.
-   * It's not currently possible at runtime to undelete a document.
-   * @returns true if the document has been marked as deleted
+   * @returns true if the handle is in one of the given states.
    */
-  isDeleted = () => this.inState([HandleState.DELETED])
-  isUnavailable = () => this.inState([HandleState.UNAVAILABLE])
   inState = (states: HandleState[]) =>
-    states.some(this.#machine?.getSnapshot().matches)
+    states.some(s => this.#machine.getSnapshot().matches(s))
 
   /** @hidden */
   get state() {
-    return this.#machine?.getSnapshot().value
+    return this.#machine.getSnapshot().value
   }
 
```

Comment:
This is marked @hidden and isn't used anywhere in this codebase - is it safe to get rid of it, or at least deprecate it?

Reply:
No, I use this often. I think we should go the other way and properly document it but I'm open to discussing further.